client/client.go

```diff
@@ -233,9 +248,30 @@ func (c *httpClusterClient) reset(eps []string) error {
 		neps[i] = *u
 	}
 
-	c.endpoints = shuffleEndpoints(c.rand, neps)
-	// TODO: pin old endpoint if possible, and rebalance when new endpoint appears
-	c.pinned = 0
+	if !c.prioritizeLeader {
+		// TODO: pin old endpoint if possible, and rebalance when new endpoint appears
+		c.endpoints = shuffleEndpoints(c.rand, neps)
+		c.pinned = 0
+	} else {
+		c.endpoints = neps
+		// TODO: should return ErrNoEndpoints in a case of getting leader fail?
```

Comment:
i think we should just fail back to no preference case. we cannot guarantee this client always connect to leader anyway. we can try our best later in the next auto sync.